core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -338,9 +344,11 @@ abstract class AbstractFetcherThread(name: String,
                     case e: KafkaStorageException =>
                       error(s"Error while processing data for partition $topicPartition", e)
                       partitionsWithError += topicPartition
```

Comment:
Should we handle this similarly as truncation and mark the partition failed?

Reply:
Yes, partitions would be now marked as failed in this case too.